### ISSUE TITLE
[Android] Cancel task

### ISSUE THREAD

**Issue by pgonzalez-santiago** [wkh237/react-native-fetch-blob#369]
Hi all, I am using fetch-blob  in my react native application and I having some problems, only in android, cancelling the download.

I have 

`  task = RNFetchBlob`
   `  .config({ path : path })` 
   `  .fetch('GET', url, {})`

`task.then((res) =>  res)`
`.catch((err) => err )`

When I run `task.cancel()` its works perfect on iOs devices returning the error, but when I do it in Android it returns the response of the 'then'.

Do you have any ideas? Thank you

RN: 0.43
Fetch-blob: ^0.10.4

**Comment by Crash--** [comment]
Just find out the same behavior

My current workaround: 
```jsx
if (this.task) {
    this.task.cancel();
    this.canceled = true;
}
///...
this.task.then((res) => {
if (res.respInfo.status == '200') {
///...
 if (this.canceled){ 
return;
}
}
}
```

**Comment by pgonzalez-santiago** [comment]
@Crash-- Thank you, that solved my problem but the issue still there.

**Comment by IceNeoMax** [comment]
i was ran into the same problem

**Comment by wkh237** [comment]
Would this relate to #268 ?

**Comment by wkh237** [comment]
Closing this issue as there's already a possible fix in #268, related discussion will continue in that thread also.
